OSDOCS11202: Clarifying and Adding Detail to "Controlling DNS pod placement” doc (#87491)
openshift/openshift-docs: modules/nw-controlling-dns-pod-placement.adoc

The DNS Operator has two daemon sets: one for CoreDNS called `dns-default` and one for managing the `/etc/hosts` file called `node-resolver`.

> You can control the nodes on which you want to assign and run CoreDNS pods. For example, if the cluster administrator has configured security policies that prohibit communication between pairs of nodes, you can restrict the set of nodes on which the daemon set for CoreDNS runs.

Author: Thanks, I got an idea on how to clear up the confusion.

DNS service is available to all pods if the following circumstances are true:

> * DNS pods are running on some nodes in the cluster.

Author: Thanks for catching this. I'll see what I can do.

.Procedure

* To allow the daemon set for CoreDNS to run on certain nodes, configure a taint and toleration:

> . Set a taint on the nodes where you want to control DNS pod placement:

Author: Thanks

+
[source,terminal]
----
$ oc adm taint nodes <node_name> dns-only=abc:NoExecute
----
+

> This command sets a taint on the specified node. Replace `<node_name>` with the actual name of the node.

Author: Agreed and removed

> . Modify the DNS Operator object named `default` to include the corresponding toleration:

Reviewer: ```suggestion
. Modify the DNS Operator object named `default` to include the corresponding toleration by entering the following command: 
```

Author: got it, thanks

.Procedure

* To allow the daemon set for CoreDNS to run on certain nodes, configure a taint and toleration:

. Set a taint on the nodes that you want to control DNS pod placement by entering the following command:
+
[source,terminal]
----

> $ oc adm taint nodes <node_name> dns-only=abc:NoExecute

Reviewer 2: ```suggestion
$ oc adm taint nodes <node_name> dns-only=abc:NoExecute <1>
```

Author: Thanks! I just didn't know if a single callout was appropriate. (Kinda like a single bullet or number in a list.) I should've asked. Sorry about that.

----
+

> Replace `<node_name>` with the actual name of the node.

Reviewer 2: ```suggestion
<1> Replace `<node_name>` with the actual name of the node.
```

Reviewer 2: I suggest adding callouts here, so its clearer that the text under the code block is important for the code above it

. Modify the DNS Operator object named `default` to include the corresponding toleration by entering the following command: 
+
[source,terminal]
----
$ oc edit dns.operator/default
----

. Specify a taint key and a toleration for the taint. The following toleration matches the taint set on the nodes.
+
[source,yaml]
----
 spec:
   nodePlacement:
     tolerations:
     - effect: NoExecute
       key: "dns-only" <1>
       operator: Equal
       value: abc
       tolerationSeconds: 3600 <2>
----
<1> If the `key` field is set to `dns-only`, it can be tolerated indefinitely.
<2> The `tolerationSeconds` field is optional.

. Optional: To specify node placement using a node selector, modify the default DNS Operator:

> ..

Reviewer 2: ```suggestion

```

Reviewer 2: ![image](https://github.com/user-attachments/assets/6d86f55e-1eeb-4e4b-bca7-4d6844edaef0)
need to remove this duplicate `..`

Author: oops! Thanks

> * To specify node placement using a node selector, modify the default DNS Operator:
> 
> . Edit the DNS Operator object named `default` to include a node selector:
> +
> [source,yaml]
> ----
>  spec:
>    nodePlacement:
>      nodeSelector:
>        node-role.kubernetes.io/control-plane: ""
> ----
> +
> This node selector ensures that the CoreDNS pods run only on control plane nodes.

Reviewer 2: ```suggestion
. Optional: To specify node placement using a node selector, modify the default DNS Operator:

..
.. Edit the DNS Operator object named `default` to include a node selector:
+
[source,yaml]
----
 spec:
   nodePlacement:
     nodeSelector:    <1>
       node-role.kubernetes.io/control-plane: ""
----
+
<1> This node selector ensures that the CoreDNS pods run only on control plane nodes.
```

Reviewer 2: Just a formatting suggestion, but I think that this should be a numbered step with an `Optional:`. As a bulleted item, it seems sorta orphaned. Also, another callout for the node selector reference.

     tolerations:
     - effect: NoExecute
       key: "dns-only"
       operator: Equal
       value: abc
       tolerationSeconds: 3600
----
+

> This toleration matches the taint set on the nodes. If the taint is `dns-only`, it can be tolerated indefinitely. You can omit `tolerationSeconds`.

Reviewer: This isn't necessary, but I guess I'm just trying to think of a way that we can use call outs. It's just something to think about. 

Might be worth considering the following. This way we can provide callouts for dns-only and  keep

```suggestion
. Specify a taint key and a toleration for the taint. The following toleration matches the taint set on the nodes.
+
[source,yaml]
----
 spec:
   nodePlacement:
     tolerations:
     - effect: NoExecute
       key: "dns-only" <1>
       operator: Equal
       value: abc
       tolerationSeconds: 3600 <2>
----
<1> If set to `dns-only`, it can be tolerated indefinitely.
<2> Optional. <Description> 
```

> * To specify node placement using a node selector, modify the default DNS operator:

Reviewer: ```suggestion
* To specify node placement using a node selector, modify the default DNS Operator:
```
